File: lib/java/src/main/java/uapi/internal/types/UBoolean.java
package uapi.internal.types;

import static uapi.internal.generation.GenerateRandomBoolean.generateRandomBoolean;
import static uapi.internal.validation.ValidateBoolean.validateBoolean;

import java.util.List;

import uapi.internal.generation.GenerateContext;
import uapi.internal.validation.ValidateContext;
import uapi.internal.validation.ValidationFailure;

public class UBoolean implements UType {

    public static final String _BOOLEAN_NAME = "Boolean";

    @Override
    public int getTypeParameterCount() {
        return 0;
    }

    @Override
    public List<ValidationFailure> validate(Object value, List<UTypeDeclaration> typeParameters, ValidateContext ctx) {
        return validateBoolean(value);
    }

    @Override
    public Object generateRandomValue(GenerateContext ctx) {
        return generateRandomBoolean(ctx);
    }

    @Override
    public String getName() {
        return _BOOLEAN_NAME;
    }

}

File: lib/py/uapi/internal/types/UBoolean.py
from typing import TYPE_CHECKING


if TYPE_CHECKING:
    from uapi.internal.validation.ValidateContext import ValidateContext
    from uapi.RandomGenerator import RandomGenerator
    from uapi.internal.validation.ValidationFailure import ValidationFailure
    from uapi.internal.types.UTypeDeclaration import UTypeDeclaration
    from uapi.internal.generation.GenerateContext import GenerateContext

from uapi.internal.types.UType import UType

_BOOLEAN_NAME: str = "Boolean"


class UBoolean(UType):

    def get_type_parameter_count(self) -> int:
        return 0

    def validate(self, value: object,
                 type_parameters: list['UTypeDeclaration'], ctx: 'ValidateContext') -> list['ValidationFailure']:
        from uapi.internal.validation.ValidateBoolean import validate_boolean
        return validate_boolean(value)

    def generate_random_value(self, ctx: 'GenerateContext') -> object:
        from uapi.internal.generation.GenerateRandomBoolean import generate_random_boolean
        return generate_random_boolean(ctx)

    def get_name(self) -> str:
        return _BOOLEAN_NAME
